src/components/PostDetails.tsx

```diff
@@ -5,31 +5,39 @@
 import * as commentsApi from '../api/comments';
 
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { actions as commentsActions } from '../features/comments/comments';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
   const [visible, setVisible] = useState(false);
 
+  const dispatch = useAppDispatch();
+  const { items, loaded, hasError } = useAppSelector(state => state.comments);
+
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    dispatch(commentsActions.setCommentLoader(false));
+    dispatch(commentsActions.setCommentError(false));
     setVisible(false);
 
     commentsApi
       .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+      .then(commentsFromServer => {
+        dispatch(commentsActions.setComments(commentsFromServer));
+      })
+      .catch(() => {
+        dispatch(commentsActions.setCommentError(true));
+      })
+      .finally(() => {
+        dispatch(commentsActions.setCommentLoader(true));
```

Comment:
The loader is set to `true` in the `finally` block, which might be incorrect. Typically, you want to set the loader to `false` to indicate loading has finished.